src/main/java/org/folio/service/holdings/HoldingsServiceImpl.java

```diff
@@ -131,8 +133,8 @@ private int getRequestCount(Integer totalCount) {
     return context.getLoadingService().getLoadingStatus();
   }
 
-  private CompletableFuture<Void> saveHolding(List<Holding> holdings, String tenantId) {
-    List<DbHolding> dbHoldings = holdings.stream()
+  private CompletableFuture<Void> saveHoldings(List<Holding> holdings, Instant updatedAt, String tenantId) {
+    Set<DbHolding> dbHoldings = holdings.stream()
```

Comment:
I'm still worried that there will be holding objects with same id, but different other attributes,
I think it's better to remove duplicates by using only id.
something like this 
```
  private CompletableFuture<Void> saveHoldings(List<Holding> holdings, Instant updatedAt, String tenantId) {
    Set<DbHolding> dbHoldings = holdings.stream()
      .filter(distinctByKey(this::getHoldingsId))
      .map(holding -> new DbHolding(
        holding.getTitleId(),
        holding.getPackageId(),
        holding.getVendorId(),
        holding.getPublicationTitle(),
        holding.getPublisherName(),
        holding.getResourceType()
      ))
      .collect(Collectors.toSet());
    logger.info("Saving holdings to database.");
    return holdingsRepository.saveAll(dbHoldings, updatedAt, tenantId);
  }

  private  <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
    Set<Object> seen = ConcurrentHashMap.newKeySet();
    return t -> seen.add(keyExtractor.apply(t));
  }

  private String getHoldingsId(Holding holding) {
    return holding.getVendorId() + "-" + holding.getPackageId() + "-" + holding.getTitleId();
  }
```

Reply:
thanks @andrii-paias, added